packages/customize-widgets/src/components/inserter/use-inserter.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { focus } from '@wordpress/dom';
+import { ESCAPE } from '@wordpress/keycodes';
+
+const {
+	wp: { customize },
+} = window;
+
+export const inserterId = 'widgets-inserter';
+
+const OuterSection = customize.OuterSection;
+// Override the OuterSection class to handle multiple outer sections.
+// It closes all the other outer sections whenever one is opened.
+// The result is that at most one outer section can be opened at the same time.
+customize.OuterSection = class extends OuterSection {
+	onChangeExpanded( expanded, args ) {
+		if ( expanded ) {
+			customize.section.each( ( section ) => {
+				if (
+					section.params.type === 'outer' &&
+					section.id !== this.id
+				) {
+					if ( section.expanded() ) {
+						section.collapse();
+					}
+				}
+			} );
+		}
+
+		return super.onChangeExpanded( expanded, args );
+	}
+};
+// Handle constructor so that "params.type" can be correctly pointed to "outer".
+customize.sectionConstructor.outer = customize.OuterSection;
+
+export const useInserterContainer = () =>
```

Comment:
This looks like a hook because it starts with `use`. Maybe rename to `getInserterContainer`?

Reply:
I intentionally name it a hook even though the implementation isn't one. So that we can hide the implementation detail in case when we have to add some states or React-only thingy in it. It's just a preference of mine though.

Reply:
I'd lean towards [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) myself but no worries I trust you!